bundles/sirix-core/src/main/java/org/sirix/service/json/serialize/JsonSerializer.java

```diff
@@ -207,11 +200,15 @@ public void emitNode(final JsonNodeReadOnlyTrx rtx) {
 
             if (withMetaData) {
               appendObjectSeparator();
-              appendObjectKeyValue(quote("hash"), quote(String.format("%032x", rtx.getHash())));
-              appendObjectSeparator().appendObjectKeyValue(quote("type"), quote(rtx.getKind().toString()))
-                                     .appendObjectSeparator()
-                                     .appendObjectKeyValue(quote("descendantCount"),
-                                                           String.valueOf(rtx.getDescendantCount()));
+              if (rtx.getHash() != null) {
```

Comment:
hm, yeah, maybe I can change this as well. I had to address several issues, as well, because the complexity of the JsonSerializer is crazy... we now allow to specify a `maxChildren` per node param to skip siblings after traversing the maximum number of children, a `maxNumberOfNodes` param to terminate the serialization process as well as a `maxDepth` param. These params are needed to visually explore the resources (currently a tree-view in the Svelte based web GUI). We could also use these for space filling approaches as TreeMaps, Icicle plots or Sunburst diagrams.